truffle/com.oracle.truffle.api/src/com/oracle/truffle/api/source/SourceSection.java

```diff
@@ -68,7 +72,7 @@
      * @param charIndex the 0-based index of the first character of the section
      * @param charLength the length of the section in number of characters
      */
-    SourceSection(String kind, Source source, String identifier, int startLine, int startColumn, int charIndex, int charLength) {
+    SourceSection(String kind, Source source, String identifier, int startLine, int startColumn, int charIndex, int charLength, String[] tags) {
```

Comment:
Missing:  discussion of equality, see `hashcode()` and `equals()` overrides.  How should the addition of tags to the object's state (which are now ignored for equality) affect use cases for `SourceSection`